Randomizer.SMZ3/Regions/Zelda/PalaceOfDarkness.cs

```diff
@@ -17,7 +17,7 @@ class PalaceOfDarkness : Z3Region, IReward {
                 new Location(this, 256+121, 0x1EA5B, LocationType.Regular, "Palace of Darkness - Shooter Room"),
                 new Location(this, 256+122, 0x1EA37, LocationType.Regular, "Palace of Darkness - Big Key Chest",
                     items => items.KeyPD >= (GetLocation("Palace of Darkness - Big Key Chest").ItemIs(KeyPD, World) ? 1 :
-                        config.Keysanity || items.Hammer && items.Bow && items.Lamp ? 6 : 5))
+                        (items.Hammer && items.Bow && items.Lamp) || config.Keysanity ? 6 : 5))
```

Comment:
So this is beyond the first time this is reverted. My thinking is that config flags should be checked first, shortcutting progression checks. It is true these checks are all constant time now, but will also be a matter of consistency with the future Z3 glitch logic, and possible SM flag guarded logic.

There's been no discussion on this logic formatting, and little on other logic expressions. Just discord me if anything comes up.